packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -201,6 +201,13 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
                     this.model.selectNode(firstChild);
                 }
             }
+        } else {
+            // On activate we have to "re-select" the nodes even if they were already selected.
+            // This is required because when toggling the side-bars (e.g. `Files`), we want to
+            // propagate the selection change to the global `SelectionService` by broadcasting
+            // the most recent tree selection state.
+            // https://github.com/theia-ide/theia/issues/2178
+            this.model.reconcileSelectionState();
```

Comment:
Why not override `onActivateRequest` in `FileNavigatorWidget` and set explicitly the global selection as for the editor widget: https://github.com/theia-ide/theia/blob/79f56f142956c7fbdf378b993bf49d24db27b935/packages/editor/src/browser/editor-widget.ts#L45-L49

Is it supposed to fix generically something else?